src/apps/invitations/api.py

```diff
@@ -233,14 +232,14 @@ async def invitation_accept(
 ):
     """General endpoint to approve the applet invitation."""
     async with atomic(session):
-        service = InvitationsService(session, user)
-        invitation = await service.get(key)
-        if invitation and invitation.role == Role.RESPONDENT:
-            if isinstance(invitation.meta, RespondentMeta):
-                subject_id = invitation.meta.subject_id
-                await SubjectsService(session, user.id).merge(
-                    uuid.UUID(subject_id)
-                )
+        invitation_srv = InvitationsService(session, user)
+        subject_srv = SubjectsService(session, user.id)
+        invitation = await invitation_srv.get(key)
+        assert invitation
```

Comment:
Invitation can be null? So you need to raise exception with appropriate message, not http 500